nist/int.go

```diff
@@ -37,8 +39,9 @@ var two = big.NewInt(2)
 // whose target is assumed never to change.
 //
 type Int struct {
-	V big.Int  // Integer value from 0 through M-1
-	M *big.Int // Modulus for finite field arithmetic
+	V          big.Int          // Integer value from 0 through M-1
+	M          *big.Int         // Modulus for finite field arithmetic
+	endianness binary.ByteOrder // Endianness considered for this int
```

Comment:
Maybe rename the element 'bo' or 'byteOrder' since that seems more self-explanatory (and more like a word) than "endianness"?

And perhaps just make it public (e.g., 'BO'), since the other elements are already public and I don't see any obvious harm/risk if this one is public too?

Finally, I'm not totally sure 'binary.ByteOrder' is the appropriate type to use here, since that's a whole interface-pointer that could in principle point to any object implementation that interface, but here the semantics are that we *need* it to be 'BigEndian' or 'LittleEndian' and nothing else.  i.e., if the caller creates their own 'myLittleEndian' object satisfying the ByteOrder interface and behaving exactly like LittleEndian, but a different pointer from LittleEndian, things will break, which violates the way interfaces are "supposed" to work.  So here I think the cleaner approach would actually be to declare a local ByteOrder type, e.g.:

type ByteOrder bool
const (
   LittleEndian ByteOrder = false
   BigEndian ByteOrder = true
)

...and just use that.